eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Proposals.hs

```diff
@@ -359,9 +365,20 @@ instance EraPParams era => EncCBOR (Proposals era) where
 instance EraPParams era => DecCBOR (Proposals era) where
   decCBOR = decCBOR >>= uncurry mkProposals
 
--- TODO: Implement Sharing: https://github.com/intersectmbo/cardano-ledger/issues/3486
 instance EraPParams era => DecShareCBOR (Proposals era) where
-  decShareCBOR _ = decCBOR
+  type
+    Share (Proposals era) =
+      ( Interns (Credential 'Staking)
+      , Interns (KeyHash 'StakePool)
+      , Interns (Credential 'DRepRole)
+      , Interns (Credential 'HotCommitteeRole)
+      )
+  decSharePlusCBOR = do
+    decodeRecordNamedT "Proposals" (const 2) $ do
+      gaid <- lift decCBOR
+      (_, omap) <- decodeListLikeWithCountT (lift decodeListLenOrIndef) (flip (OMap.|>)) $ \_ ->
+        decSharePlusCBOR
+      mkProposals gaid omap
```

Comment:
How cool that we can reuse this here thanks to MonadFail :star_struck: 

Wondering like before if anything is added to the shared state here, maybe the `GovActionState`  because we're using `decSharePlusCBOR` for it?

Reply:
The call to `decSharePlusCBOR` in:
```
      (_, omap) <- decodeListLikeWithCountT (lift decodeListLenOrIndef) (flip (OMap.|>)) $ \_ ->
        decSharePlusCBOR
```

will add things because the decoder for each `GovActionState` does it (see the `put` call).